src/coreclr/vm/ceeload.cpp

```diff
@@ -4734,18 +4765,20 @@ VASigCookie *Module::GetVASigCookie(Signature vaSignature)
 
         // Compute the size of args first, outside of the lock.
 
-        // @TODO GENERICS: We may be calling a varargs method from a
-        // generic type/method. Using an empty context will make such a
-        // case cause an unexpected exception. To make this work,
-        // we need to create a specialized signature for every instantiation
-        SigTypeContext typeContext;
-
-        MetaSig metasig(vaSignature, this, &typeContext);
+        MetaSig metasig(vaSignature, this, typeContext);
         ArgIterator argit(&metasig);
 
         // Upper estimate of the vararg size
         DWORD sizeOfArgs = argit.SizeOfArgStack();
 
+        // Prepare instantiation
+        LoaderAllocator * pAllocator = pLoaderModule->GetLoaderAllocator();
+
+        DWORD classInstCount = typeContext->m_classInst.GetNumArgs();
+        DWORD methodInstCount = typeContext->m_methodInst.GetNumArgs();
+        pAllocator->EnsureInstantiation(this, typeContext->m_classInst);
+        pAllocator->EnsureInstantiation(this, typeContext->m_methodInst);
+
         // enable gc before taking lock
         {
             CrstHolder ch(&m_Crst);
```

Comment:
Does this need to be from `pLoaderModule` as well? 

It may be better to move everything that works against `pLoaderModule` into a separate method where the loader module is `this`. It would prevent mistakes like this one.

Reply:
Done.